esp/esp/program/modules/handlers/schedulingcheckmodule.py

```diff
@@ -47,71 +47,72 @@ def format_table(self, l, options={}, help_text=""):
     def format_list(self, l, options={}, help_text=""):
         return l
 
-class HTMLSCFormatter:
+# Builds JSON output for an object with attributes help_text, headings, and body.
+class JSONFormatter:
     #requires: d, a two level dictionary where the the first set of
     #   keys are the headings expected on the side of the table, and
     #   the second set are the headings expected on the top of the table
     def format_table(self, d, options={}, help_text=""):
         if isinstance(d, list):
-            return self._format_list_table(d, options['headings'], help_text=help_text)
+            return json.dumps(self._format_list_table(d, options['headings'], help_text=help_text))
         else:
-            return self._format_dict_table(d, options['headings'], help_text=help_text)
+            return json.dumps(self._format_dict_table(d, options['headings'], help_text=help_text))
 
-    def format_list(self, l, help_text=""):
-        output = self._table_start(help_text)
+    def format_list(self, l, help_text=""): # needs verify
+        output = {}
+        output["help_text"] = help_text
+        output["headings"] = [] # no headings
+        
+        # might be redundant, but it makes sure things aren't in a weird format
+        body = []        
         for row in l:
-            output += self._table_row([row])
-        output += "</table>"
-        return output
-
-    def _table_start(self, help_text=""):
-        output = ''
-        if help_text:
-             output += '<div class="help-text">%s</div>' % help_text
-        output += '<table cellpadding=10>'
-        return output
+            body.append(self._table_row([row]))
+        output["body"] = body
+        return json.dumps(output)
 
-    def _table_headings(self, headings):
+    def _table_headings(self, headings): # in case headings are a dict
         #column headings
-        next_row = ""
+        next_row = []
         for h in headings:
-            next_row = next_row + "<th><div style=\"cursor: pointer;\">" + str(h) + "</div></th>"
-        next_row = next_row + "</tr></thread>"
+            next_row.append(str(h))
         return next_row
 
     def _table_row(self, row):
-        next_row = ""
+        next_row = []
         for r in row:
             #displaying lists is sometimes borked.  This makes it not borked
             if isinstance(r, list):
                 r = [str(i) for i in r]
-            next_row += "<td>" + str(r) + "</td>"
-        next_row += "</tr>"
+            next_row.append(str(r))
```

Comment:
Empirically, no (I think e.g. lunch blocks is an example where you would expect that to happen). I think it's because javascript is rendering each cell separately.